module/VuFind/src/VuFind/RecordDriver/AbstractBase.php

```diff
@@ -454,6 +454,16 @@ public function getExtraDetail($key)
         return $this->extraDetails[$key] ?? null;
     }
 
+    /**
+     * Return the collection search ID for this record.
+     *
+     * @return string
+     */
+    public function getCollectionSearchId(): string
```

Comment:
Other collection-related functionality is found in the [HierarchyAwareTrait](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/RecordDriver/Feature/HierarchyAwareTrait.php). I wonder if it would make more sense to move this there.